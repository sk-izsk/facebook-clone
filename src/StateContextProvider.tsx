import React, { createContext, useContext, useReducer } from 'react';

export interface StateContextProviderProps {
  initialState?: any;
  reducer?: any;
}

const StateContext = createContext(undefined);

const StateContextProvider: React.FC<StateContextProviderProps> = ({ initialState, reducer, children }) => {
  return <StateContext.Provider value={useReducer(reducer, initialState) as any}>{children}</StateContext.Provider>;
};

const useStateValue: any = () => useContext(StateContext);

export { StateContextProvider, StateContext, useStateValue };
