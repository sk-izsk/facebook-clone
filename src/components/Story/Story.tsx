import { Avatar, Box, makeStyles, Typography } from '@material-ui/core';
import React from 'react';
import { CustomTheme } from '../../theme/muiTheme';

export interface StoryProps {
  image?: any;
  profileSrc?: string;
  title?: string;
}

const useStyles = makeStyles((theme: CustomTheme) => ({
  story: {
    position: 'relative',
    backgroundPosition: 'center center',
    backgroundSize: 'cover',
    backgroundRepeat: 'no-repeat',
    width: 120,
    height: 200,
    boxShadow: `0px 5px 17px -7px ${theme.palette.primary.dark}`,
    borderRadius: 10,
    marginRight: theme.spacing(1.2),
    transition: 'transform 100ms ease-in',
    cursor: 'pointer',
    '&:hover': {
      transform: 'scale(1.07)',
    },
  },
  title: {
    color: theme.palette.primary.light,
    position: 'absolute',
    bottom: theme.spacing(2.5),
    left: theme.spacing(2.5),
  },
  storyAvatar: {
    margin: theme.spacing(1.2),
    border: `5px solid ${theme.palette.primary.main}`,
  },
}));

const Story: React.FC<StoryProps> = ({ image, profileSrc, title }) => {
  const classes = useStyles();
  return (
    <Box style={{ backgroundImage: `url(${image})` }} className={classes.story}>
      <Avatar className={classes.storyAvatar} src={profileSrc} />
      <Typography className={classes.title} variant='h6'>
        {title}
      </Typography>
    </Box>
  );
};

export { Story };
