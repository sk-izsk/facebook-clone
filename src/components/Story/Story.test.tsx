import { render } from '@testing-library/react';
import React from 'react';
import { StateContextProvider } from '../../StateContextProvider';
import { Story } from './Story';

const mockUser = {
  image: 'fake image url',
  message: 'fake message',
  profilePic: 'fake profile pic',
  timestamp: 'fake time stamp',
  username: 'fake username',
};

test('Story renders correctly', () => {
  const { asFragment } = render(
    <StateContextProvider initialState={{ user: mockUser }}>
      <Story />
    </StateContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
