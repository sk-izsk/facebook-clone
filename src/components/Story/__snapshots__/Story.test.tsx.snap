// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-4 makeStyles-story-1"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circle makeStyles-storyAvatar-3 MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiAvatar-fallback"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
    <h6
      class="MuiTypography-root makeStyles-title-2 MuiTypography-h6"
    >
      mock title
    </h6>
  </div>
</DocumentFragment>
`;
